### ISSUE TITLE
How does the bot authenticate against JudgeApps?

### ISSUE THREAD

**Issue by bra1n** [Takeno/magic-judges-event-bot#7]
Hi, author of the [judgebot](https://github.com/bra1n/judgebot) here!
We've been looking for a way to add event listing from Judgeapps to our bot, too, but so far it didn't seem like there would be an API available for public use. Now given that your bot promises to solve this problem, I was wondering how you handled this issue. Looking at the code only showed me that you made a request against https://apps.magicjudges.org/events/ without any authentication, but trying to do this with an incognito browser tab only bounced me to the login page. Am I doing something wrong?
Thanks,
Steffen

**Comment by Takeno** [comment]
Hi @bra1n ! 

This script was parsing the html page because no API was available. I found that https://apps.magicjudges.org/events/  *was* public without authentication.

We noticed that it stop to work some days ago but we changed our channels these days and I thought that the problem was on webhook. (Yeah, tests are missing).

It seems they changed the security policy for that page. I'll investigate about possible solutions.

Thank you for the feedback!
Matteo

**Comment by bra1n** [comment]
That's interesting, because when we looked into adding this feature to judgebot last year, the page was already requiring authentication. So it seems like you created your bot in the "brief" window that had the page accessible without a login. :roll_eyes: If only they already added an API...

**Comment by Takeno** [comment]
I think so! We used this bot for months :D

**Comment by Takeno** [comment]
Actually it seems that the only public (aka non-protected) API is the search:
https://apps.magicjudges.org/api/events/?q=Grand%20Prix

**Comment by bra1n** [comment]
Huh, neat. Is there a documentation for it somewhere?

**Comment by Takeno** [comment]
Nope, of course. This is the _crawler adventures_. I've found it inspecting the network.

**Comment by bra1n** [comment]
Meh, not surprised there... :( Looks like it is an internal API, too. So it's doubtful that it will ever be useful for anything but really really basic event searching.

**Comment by pnicolli** [comment]
Cool part about this is that is also shows events that are not public:
Compare it with this https://apps.magicjudges.org/events/?filter_type=2
You can see events from the api which are probably being prepared at this time, but not published yet. For example, september GPs.

We should probably use this as another example to show that JudgeApps could really be more powerful if open sourced.

**Comment by bra1n** [comment]
Interesting. That makes it almost certain that this API is not intended for public use and will probably be walled off against unauthenticated users soon, too. I've been a part of the JudgeApps development team for a while and while they know that open source would be better, it is very unlikely to ever happen, due to some rather "sensitive" code. (\*cough\*Exemplar\*cough\*)

**Comment by Takeno** [comment]
[OT]I don't know what you mean with "sensitive" code. Is it protected business logic? I want to contribute on how exemplar are shown, not read recognitions before they are public.

Btw, I've open a "judgeapp project" about open source developments for judges from judges, if you are interested.[/OT]

Back to this issue.. I think actually the only way is an authorization request before parse events, and this need a valid account with credentials.

**Comment by bra1n** [comment]
> I don't know what you mean with "sensitive" code. Is it protected business logic? I want to contribute on how exemplar are shown, not read recognitions before they are public.

That's the insanity of it - it is purely kept closed source so that people can't "game the system". Which - in my opinion - is the same bulls*** as "security through obscurity". Essentially, there are only a couple of people involved in the core development process of both Apps and Exemplar and however they decide to do things is how things get done. I've tried arguing about this with them, but to no avail. That's one of the reasons I stopped working on most of the judge projects.

> Back to this issue.. I think actually the only way is an authorization request before parse events, and this need a valid account with credentials.

Right, but I don't want to have my personal account associated with automatic access to the database. This is not what my account is supposed to be used for and in the long term will probably get me suspended. Creating accounts just for bots is also not allowed either, for the same reasons. I mean, you could simply do it anyway and nobody would know unless you started requesting the event list every other second, but I'd rather not risk my judge level for that. ;)

**Comment by Takeno** [comment]
> That's the insanity of it - it is purely kept closed source so that people can't "game the system". Which - in my opinion - is the same bulls*** as "security through obscurity". Essentially, there are only a couple of people involved in the core development process of both Apps and Exemplar and however they decide to do things is how things get done. I've tried arguing about this with them, but to no avail. That's one of the reasons I stopped working on most of the judge projects.

I know that situation. That's why I want to keep open every community project like this one. With @pnicolli we started to pulish projects under https://github.com/imj

>Right, but I don't want to have my personal account associated with automatic access to the database. This is not what my account is supposed to be used for and in the long term will probably get me suspended. Creating accounts just for bots is also not allowed either, for the same reasons. I mean, you could simply do it anyway and nobody would know unless you started requesting the event list every other second, but I'd rather not risk my judge level for that. ;)

I will contact JudgeApps' support to find a legal way.

Thank you!
